json_rpc/http.md

```diff
@@ -0,0 +1,243 @@
+
+In descending order of commitment (most finalized to least finalized), these are the commitment levels:
+
+- `"finalized"` - the node will query the most recent block confirmed by supermajority
+  of the cluster as having reached maximum lockout, meaning the cluster has
+  recognized this block as finalized
+- `"confirmed"` - the node will query the most recent block that has been voted on by supermajority of the cluster.
+  - It incorporates votes from gossip and replay.
+  - It does not count votes on descendants of a block, only direct votes on that block.
+  - This confirmation level also upholds "optimistic confirmation" guarantees in
+    release 1.3 and onwards.
+- `"processed"` - the node will query its most recent block. Note that the block
+  may still be skipped by the cluster.
+
+Please refer to [configuring-state-commitment](https://docs.solana.com/api/http#configuring-state-commitment) for more details.
+
+#### RpcResponse Structure
+
+Many methods that take a commitment parameter return an RpcResponse JSON object comprised of two parts:
+
+- `context` : An RpcResponseContext JSON structure including a `slot` field at which the operation was evaluated. example:
+
+  ```json
+      "context": {
+      "slot": 1
+    },
+  ```
+
+- `value` : The value returned by the operation itself. example:
+
+```json
+        "value": [
+            {
+                "bundle": {
+                    "signatures": [
+                        "3Eq21vXNB5s86c62bVuUfTeaMif1N2kUqRPBmGRJhyTA",
+                        "2nBhEBYYvfaAe16UMNqRHre4YNSskvuYgx3M6E4JP1oDYvZEJHvoPzyUidNgNX5r9sTyN1J9UxtbCXy2rqYcuyuv"
+                    ]
+                },
+                "slot": 1234,
+                "confirmationStatus": "finalized",
+                "err": "null"
+            }
+        ]
+```
+
+## JSON RPC API Reference
+
+## getTipAccounts
+
+Returns the tip accounts for tip payment for the bundles.
+
+### Parameters
+
+None
+
+### Result
+
+The result field will be a JSON object with the following fields:
+
+- `result`: `<array>` - Tip accounts as a list of `strings`
+
+### Code sample
+
+#### Request
+
+```bash
+curl https://mainnet.block-engine.jito.wtf:443/api/v1/bundles -X POST -H "Content-Type: application/json" -d '
+{
+    "jsonrpc": "2.0",
+    "id": 1,
+    "method": "getTipAccounts",
+    "params": []
+}
+'
+```
+
+#### Response
+
+```json
+{
+    "jsonrpc": "2.0",
+    "result": [
+        "9n3d1K5YD2vECAbRFhFFGYNNjiXtHXJWn9F31t89vsAV",
+        "aTtUk2DHgLhKZRDjePq6eiHRKC1XXFMBiSUfQ2JNDbN",
+        "B1mrQSpdeMU9gCvkJ6VsXVVoYjRGkNA7TtjMyqxrhecH",
+        "9ttgPBBhRYFuQccdR1DSnb7hydsWANoDsV3P9kaGMCEh",
+        "4xgEmT58RwTNsF5xm2RMYCnR1EVukdK8a1i2qFjnJFu3",
+        "EoW3SUQap7ZeynXQ2QJ847aerhxbPVr843uMeTfc9dxM",
+        "E2eSqe33tuhAHKTrwky5uEjaVqnb2T9ns6nHHUrN8588",
+        "ARTtviJkLLt6cHGQDydfo1Wyk6M4VGZdKZ2ZhdnJL336"
+    ],
+    "id": 1
+}
+```
+
+## sendBundle
```

Comment:
we should add a section on how to bid/tip for the bundle inclusion. a tip can be any instruction, top-level or CPI, that transfers SOL to one of the 8 tip accounts

Reply:
clients should make sure they have balance and state assertions that allow the tip to only go thru conditionally, especially if tipping as a separate tx

Reply:
Added 
`
Please note that a tip is necessary for the bundle to considered. A tip can be any instruction, top-level or CPI, that transfers SOL to one of the 8 tip accounts. Clients should make sure they have balance and state assertions that allow the tip to only go thru conditionally, especially if tipping as a separate tx. If the tip is low, there is a chance that the bundle does not get selected during the auction. You can get the tip accounts using [getTipAccounts](#gettipaccounts). Ideally select one of the accounts in random to reduce contention.
`